### PR TITLE
[TC-6] WIP adds non-interactive mode to traffic_ops postinstall

### PR DESCRIPTION
**Pull request by PeterRyder**
Initial PR for non-interactive traffic ops postinstall. Along with changing the postinstall file, some of the helper files such as build_trafficops_perl_library and generateCerts have been moved to /lib and made into perl modules

### CODE REVIEW

File: traffic_ops/install/bin/postinstall-new [hunk 1]
```diff
@@ -0,0 +1,781 @@
+#!/usr/bin/perl
```
Comment (dangogh) [inline traffic_ops/install/bin/postinstall-new]: Go ahead and replace the postinstall script that's there -- we have it in the git history.   That will allow us to review the changes directly without having to diff offline.

File: traffic_ops/install/bin/postinstall-new [hunk 2]
```diff
@@ -0,0 +1,781 @@
...
+use BuildPerlDeps qw{ :all };
+use GenerateCert qw{ :all };
+use ProfileCleanup qw { :all };
+use Digest::SHA1 qw(sha1_hex);
+use Data::Dumper qw(Dumper);
+use Scalar::Util qw(looks_like_number);
+use Getopt::Long;
+
+# paths of the output configuration files
+our $databaseConfFile = "/opt/traffic_ops/app/conf/production/database.conf";
```
Comment (dangogh) [inline traffic_ops/install/bin/postinstall-new]: No reason to use "our" -- here -- sharing package variables makes it hard to follow the code.

File: traffic_ops/build/traffic_ops.spec [hunk 3]
```diff
@@ -122,7 +122,7 @@ Built: %(date) by %{getenv: USER}
     # install
     if [ "$1" = "1" ]; then
       # see postinstall, the .reconfigure file triggers init().
-      /bin/touch %{PACKAGEDIR}/.reconfigure
+      #/bin/touch %{PACKAGEDIR}/.reconfigure
```
Comment (PeterRyder) [inline traffic_ops/build/traffic_ops.spec]: Removed because reconfigure is now a command line option - removed line from spec file

File: traffic_ops/install/bin/input.json [hunk 4]
```diff
@@ -67,5 +67,102 @@
...
+    {
+      "Organizational Unit Name (eg, section)":"",
+      "config_var":"org_unit"
+    },
+    {
+      "Common Name (eg, your name or your server's hostname)":"example.com",
+      "config_var":"common_name"
+    },
+    {
+      "RSA Passphrase":"password",
```
Comment (dewrich) [inline traffic_ops/install/bin/input.json]: Passwords in files always worry me, we should maybe evaluate a command line option or have a separate file that can be stored locally to be safe.
